### PR TITLE
Support multi-versioned mgmt network compute resources SDK packages

### PR DESCRIPTION
**Pull request by derekbekoe**
It still only works with the api versions for public Azure.
The required changes haven’t been made to support other API versions.

### CODE REVIEW

**Comment by derekbekoe** [comment]
@lmazuel for changes to `shared.py`

**Comment by derekbekoe** [comment]
@tjprescott @yugangw-msft @troydai Can someone from the CLI team take a look also?

File: src/azure-cli-core/azure/cli/core/profiles/shared.py [hunk 1]
```diff
@@ -78,7 +133,15 @@ def get_client_class(resource_type):
...
-    return getattr(import_module(resource_type.sdk_module), resource_type.client_name)
+    cp = resource_type.client_path
+    try:
+        mod_to_import, attr_path = cp.split('#')
+        op = import_module(mod_to_import)
+        for part in attr_path.split('.'):
+            op = getattr(op, part)
+        return op
+    except (ValueError, AttributeError):
+        return None
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: What caller will do if get None back?

File: src/azure-cli-core/azure/cli/core/profiles/shared.py [hunk 2]
```diff
@@ -22,34 +22,89 @@ def __str__(self):
 
 class ResourceType(Enum):  # pylint: disable=too-few-public-methods
 
+    # TODO Thinking of removing the RP/RT format due to swagger/sdk format
     MGMT_STORAGE_STORAGE_ACCOUNTS = ('Microsoft.Storage/storageAccounts',
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: I suggest just call `MGMT_STORAGE`, unless I missed other storage entries

File: src/azure-cli-core/azure/cli/core/profiles/shared.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: It couldn't get the client class.
Caller shouldn't need to handle this as it shouldn't happen.
**If** it does happen, there's not much the caller can do as they needed the client for some purpose.

@yugangw-msft As an alternative, I can remove the try/except and just let the exception be uncaught. Thoughts?

Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: yep, I vote for removing the try/except

File: src/azure-cli-core/azure/cli/core/profiles/shared.py (hunk 2 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: done.

File: src/azure-cli-core/azure/cli/core/profiles/shared.py (hunk 1 above)
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/profiles/shared.py]: will go with that.

**Pull request merged**